### PR TITLE
Admin detail

### PR DESCRIPTION
**Pull request by Kailiangdong**
According to mock up issue #5. I make the overview of tickets of certain events.
User can search ticket by name, no matter uppercase or lowercase.
Scanning by FAB.

I still show the TUM-Id and date in every item of list, because if not, it seems only name will show on every list. It seems too less? We can discuss it here, what kind of thing can be added in every item

No swipeRefresh layout ,because it has conflict with scrolling down in ListView, although I know there are some ways (which is a little bit complex)to solve this conflict, I think  putting refresh button is a easier and directer way to refresh tickets.
image

### CODE REVIEW

**Comment by saileru** [comment]
Yes, I like the refresh button in the toolbar!
I think TUM-ID and dates might really not be necessary in this view. If everything is shown in this view already, the details view might be redundant...
Only name and maybe ticket type are sufficient. I would prefer to have the ticket type description instead of the id.

I will look over the code tomorrow :)

**Comment by Kailiangdong** [comment]
Add  total sale for a event Finish #4 
image

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    private SearchView searchView;
+    private ListView listView;
+    private TextView totalSaleView;
+    private FloatingActionButton floatingScanner;
+    private TicketListAdapter mAdapter;
+    private TicketListAdapter findAdapter;
+    private List<AdminTicket> tickets;
+    private List<AdminTicket> findList;
+    private MenuItem menuItemScanView;
```
Comment (saileru) [inline app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java]: This attribute is not used currently. Do we need it?

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        startActivity(new Intent(this, TicketScanActivity.class));
+        //TODO:send current eventId to TicketScanActivity
+    }
+
+    private void setupSearchView() {
+        searchView.setSubmitButtonEnabled(true);
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            public boolean onQueryTextSubmit(String query) {
+                if (TextUtils.isEmpty(query)) {
+                    Toast.makeText(AdminDetailsActivity.this, "please text something", Toast.LENGTH_SHORT).show();
```
Comment (saileru) [inline app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java]: I think we should use Utils.showToast
Also add the strings to strings.xml

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                    listView.setAdapter(mAdapter);
+                } else {
+                    findList.clear();
+                    for (int i = 0; i < tickets.size(); i++) {
+                        AdminTicket findticket = tickets.get(i);
```
Comment (saileru) [inline app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java]: Naming convention: foundTicket

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                        Toast.makeText(AdminDetailsActivity.this, "Search successfully", Toast.LENGTH_SHORT).show();
+                        findAdapter = new TicketListAdapter(findList, AdminDetailsActivity.this);
+                        listView.setAdapter(findAdapter);
+                    }
+                }
+                return true;
+            }
+
+            public boolean onQueryTextChange(String newText) {
+                if (TextUtils.isEmpty(newText)) {
```
Comment (saileru) [inline app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java]: onQueryTextSubmit and onQueryTextChange have a lot of code in common. Can we maybe extract this as a method which both methods can use?
If you have statements which should only be executed for one method you can do something like:

private void handleQuery(boolean isSubmission){
    ....
    if (isSubmission) Utils.showToast(...)
    ....
}

**Review by saileru — CHANGES_REQUESTED** [review]
Good job! Nice implementation for the search bar :)

Just a few suggestions then we can merge it.
#7 says that also the TUM-ID can be used to find a ticket. 
Two options: Either implement this in this PR or do it in a separate one

I would like to merge this one quickly, so that we can continue from the same code base.

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java (hunk 1 above)
Comment (Kailiangdong) [reply app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java]: deleted it

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java (hunk 2 above)
Comment (Kailiangdong) [reply app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java]: Done it

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java (hunk 3 above)
Comment (Kailiangdong) [reply app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java]: Done it

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java (hunk 4 above)
Comment (Kailiangdong) [reply app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java]: I find that onQueryTextSubmit and onQueryTextChange  look similar at first glance. But they have much difference in a lot of detail? I may not able to extract them in a method. Others are finished. I add function to search by Id

**Review by saileru — APPROVED** [review]

**Comment by saileru** [comment]
@kordianbruck We need help here. Apparently we are not authorized to merge into master!

**Comment by kordianbruck** [comment]
@saileru upps, sorry. Now it should work. My bad.

**Pull request merged**